source/bdDetect.py

```diff
@@ -114,50 +114,70 @@ def getDriversForConnectedUsbDevices():
 	@return: Pairs of drivers and device information.
 	@rtype: generator of (str, L{DeviceMatch}) tuples
 	"""
-	usbDevs = itertools.chain(
+	usbDevsForCustom, usbDevsForHID = itertools.tee(itertools.chain(

```

Comment:
I think I already told you my concern about this privately. Ib short, this means that we are iterating over non HID matches to find out whether they ought to be used with the HID driver. If you insist on keeping this as is, I don't really care though.

Reply:
I have attempted to address this. I wasn't too worried about this because the `intertools.tee` caches its output, meaning the list is only created once then kept in memory. By the time we get to looking for standard HID braille devices, all the matches have already been created and processed once. The second loop doesn't cause anything to be created. The test to filter to HID is quick, and the number of devices low.

However, there may be a cognitive cost to future developers who don't understand the intention. Hopefully I haven't made any mistakes while refactoring this.